src/firefly/js/charts/ui/HistogramOptions.jsx

```diff
@@ -60,6 +60,41 @@ const algorithmOptions = [  {label: 'Bayesian blocks', value: 'bayesianBlocks'},
 const binSizeOptions = [  {label: 'Number of bins:', value: 'numBins'},
     {label: 'Bin width:', value: 'binWidth'} ];
 
+
+var columnNameReducer= (colValStats, histogramParams) => {
+    if (!colValStats) {
+        return {};
+    }
+    return (inFields, action) => {
+
+        if (!inFields) {
+            return {};
+        }
+        let fieldKey = undefined;
+        if (action.type === VALUE_CHANGE) {
+            // when column name changes, update the min/max input
+            fieldKey = get(action.payload, 'fieldKey');
+            const colName = action.payload.value;
+            if (fieldKey === 'columnOrExpr') {
+
+                for (var i=0; i<colValStats.length; i++){
+                    if (colName=== colValStats[i].name){
+                        const dataMin = colValStats[i].min;
+                        const dataMax = colValStats[i].max;
+                        const binWidth = (dataMax - dataMin)/50.0;
+                        inFields = updateSet(inFields, ['minCutoff', 'value'],`${dataMin}`);
+                        inFields = updateSet(inFields, ['maxCutoff', 'value'], `${dataMax}`);
+                        inFields = updateSet(inFields, ['binWidth', 'value'], `${binWidth}`);
+                        break;
+
+                    }
+                }
```

Comment:
When colName is an expression (it can not be found in the colValStats), the values should be set to undefined. (Otherwise we have wrong suggestions, which user should remove one by one.)